Nightwatch.Tests/Configuration.fs

```diff
@@ -36,21 +35,50 @@ let ``Configuration should read the YAML file`` () =
     let text = @"version: 0.0.1.0
 id: test
 schedule: 00:05:00
-check: ping localhost"
+type: test
+param:
+    check: ping localhost"
+    let checker = ResourceChecker(fun () -> failwith "nop")
     let expected =
         { id = "test"
           runEvery = TimeSpan.FromMinutes 5.0
-          checkCommand = "ping localhost" }
+          checker = checker }
+    let mutable parsedParam = None
+    let factory : ResourceFactory =
+        { resourceType = "test"
+          create = Func<_, _>(fun param -> parsedParam <- Some param; checker) }
+    let registry = Registry.create [| factory |]
     let fileSystem = mockFileSystem [| "dir/test.yml", text |]
     async {
-        let! result = Configuration.read fileSystem (Path "dir")
+        let! result = Configuration.read registry fileSystem (Path "dir")
         Assert.Equal<_>([| Ok expected |], result)
+
+        let param = Option.get parsedParam
+        Assert.Equal("ping localhost", param.["check"])
+    } |> Async.StartAsTask
+
+let private emptyRegistry = Registry.create [| |]
+
+[<Fact>]
+let ``Confiuguration returns error if the type is not registered in the factory``() =
+    let text = @"version: 0.0.1.0
+id: test
+schedule: 00:05:00
+type: test"
+    let path = "dir/test.yml"
+    let fileSystem = mockFileSystem [| path, text |]
+    let expected = Error { path = (Path path)
+                           id = Some "test"
+                           message = "The resource factory for type \"test\" is not registered" }
```

Comment:
Consider creating issue about sharing error messages between `Core` and `Tests`.

Reply:
Currently I don't think we need that too much. If we decide we want to test every error message — then we'll share them "automatically" (i.e. without creating an additional issue).